app/modules/sjson.c

```diff
@@ -153,10 +153,41 @@ create_new_element(jsonsl_t jsn,
 }
 
 static void push_number(JSN_DATA *data, struct jsonsl_state_st *state) {
-  lua_pushlstring(data->L, get_state_buffer(data, state), state->pos_cur - state->pos_begin);
-  LUA_NUMBER r = lua_tonumber(data->L, -1);
+  const char *start = get_state_buffer(data, state);
+  const char *end = start + state->pos_cur - state->pos_begin;
+  lua_pushlstring(data->L, start, end - start);
+  // See if there is any chance that this is an integer
+#if LUA_VERSION_NUM >= 503
+  char notint = 0;
```

Comment:
@pjsg TBH Philip, the only reason that I came across this was because of my work to harmonise the 5.1 and 5.3 APIs so that we don't need to have conditional `LUA_VERSION_NUM` variant code in our modules.  In general Lua hides sub types in its exposed API, so there is no difference from an API perspective between the various function subtypes: they are all classed as function.  Ditty short vs long strings, and Tables vs ROTables.   The one exception is the handling of integer vs number subtypes as there is an `lua_isinteger()` call and variant get and put API calls.  This one is a sensible extension one one that IMO is also worth back-porting into the Lua API.  If I do this, then I will also remove this variant code as the 503 version will work fine with both versions.

Reply:
This case is ugly -- and made worse by the fact that the input buffer is not null terminated and I was nervous that the number parsing might fall off the end of the buffer... I *suspect* that it would never happen, but I prefer to err on the side of caution.